### PR TITLE
[kf5textwidgets] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**

- #### What does your PR fix?  
  Adds new kf5textwidgets port. 
  Depends on:
  - [x] #16562. Was merged to this PR for testing purposes, but please review separately. I will rebase once it gets merged upstream.
  - [x] #20016. Was merged to this PR for testing purposes, but please review separately. I will rebase once it gets merged upstream.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
   Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

### CODE REVIEW

File: ports/kf5configwidgets/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO KDE/kconfigwidgets
+    REF v5.84.0
+    SHA512 da03f4cfc2a64b3ccccfe2b6b7f392f84aba2b975edbf6a5b08a14604ccb565a4491c7eb707af7191345b55ca81e864b7ee13fe648589a56f3226c26160ed024
+    HEAD_REF master
+    PATCHES
+        71.patch
```
Comment (JonLiu1993) [inline ports/kf5configwidgets/portfile.cmake:8]: Could you please rename these patch files to briefly describe what you are fixing?

File: ports/kf5guiaddons/portfile.cmake [hunk 2]
```diff
@@ -4,6 +4,9 @@ vcpkg_from_github(
     REF v5.84.0
     SHA512 e5905c0aa5343ce3d4cd3765cb81390fc89fb78aec3c8de8b31d1dada8074d04f549ff785f3988498d2e274d7cb08a35a83ba031d18562049e6ca41d18ea52ee
     HEAD_REF master
+    PATCHES
+        25.patch # https://invent.kde.org/frameworks/kguiaddons/-/merge_requests/25
+        fix_cmake.patch # see https://github.com/microsoft/vcpkg/issues/17607#issuecomment-831518812
```
Comment (JonLiu1993) [inline ports/kf5guiaddons/portfile.cmake:9]: Could you please rename these patch files to briefly describe what you are fixing?

**Comment by wrobelda** [comment]
@JonLiu1993 this no longer depends on any PRs, all dependencies are in master. All requests solved as well.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
I'd like to see a solution to https://github.com/microsoft/vcpkg/pull/19195#pullrequestreview-744280457 before merging this

**Comment by wrobelda** [comment]
> I'd like to see a solution to [#19195 (review)](https://github.com/microsoft/vcpkg/pull/19195#pullrequestreview-744280457) before merging this

Same solution as noted in https://github.com/microsoft/vcpkg/pull/19195#discussion_r703106264

**Comment by JonLiu1993** [comment]
@wrobelda ,This pr also depends on #20090

**Comment by wrobelda** [comment]
> @wrobelda ,This pr also depends on #20090 ?

@JonLiu1993 not really, the #20016 provided all the fixes for `kf5i18n`. Once #20090 gets merged, I'll update `kf5i18n` and remove its patch.

@20016 was merged to master already, so I rebased this PR and it's good to go.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks again!

**Pull request merged**